pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -0,0 +1,90 @@
+package cosmodb
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+var _ http.RoundTripper = (*tracerRoundTripper)(nil)
+
+type tracerRoundTripper struct {
+	m   metrics.Interface
+	tr  http.RoundTripper
+	log *logrus.Entry
+}
+
+func New(log *logrus.Entry, tr *http.Transport, m metrics.Interface) *tracerRoundTripper {
+	return &tracerRoundTripper{
+		m:   m,
+		tr:  tr,
+		log: log,
+	}
+}
+
+func (t *tracerRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := t.tr.RoundTrip(r)
+	if resp == nil {
+		return resp, err
+	}
+
+	action := "unknown"
+	table := "unknown"
+	code := "unknown"
+	ru := float64(0)
+
+	url := r.URL.String()
+	// extract last path element: example.com/dbs/dbname/colls/OpenShiftClusters/{docs}
+	ld := strings.LastIndexByte(url, '/')
+	action = url[ld+1:]
+	// extract last path element from prev ref point: example.com/dbs/dbname/colls/{OpenShiftClusters}
+	table = url[strings.LastIndexByte(url[:ld], '/')+1:]
+
+	code = strconv.Itoa(resp.StatusCode)
+
+	// Sometimes we get request-charge="" because pkranges API is free
+	// We log this on debug mode only and ignore
+	if len(resp.Header.Get("x-ms-request-charge")) > 0 {
+		f, err := strconv.ParseFloat(strings.Trim(resp.Header.Get("x-ms-request-charge"), "\""), 64)
+		if err != nil {
+			// we don't want to kill all DB call if this changes
```

Comment:
I am OK with returning the error and finding out about the brokenness sooner

Reply:
I think we should have "Logs to Metrics" and act on this. This goes to the topic "Will this wake me at 3 in the morning"

Reply:
ok